src/util/indexrange.h

```diff
@@ -140,6 +140,8 @@ class IndexRange final: private std::pair<SINT, SINT> {
     // the length of this range.
     IndexRange splitAndShrinkBack(SINT backLength);
 
+    bool isPartOf(IndexRange outerIndexRange) const;
```

Comment:
The term "part" usually refers to discrete components which does no apply here. This is a *containment* relationship, not a *whole-part* relationship. The term *contains* would fit here, returning the inverse of *isPartOf*.

Reply:
Maybe isSectionOf, isSubSectionOf or isSegmentOf ?

Reply:
Section will be a new therm compared to Range, so I think we want isSubrangeOf()
But I am actually more leaning to isContainedIn()

Reply:
Both work for me. I would prefer `isSubrangeOf()` because *containment* is often used in the context of collections.

Reply:
:+1 for introducing as few terms as possible. Therefore *subrange* fits very well here.

@Be is *subrange* a commonly used term? Because a single word would be easier to grasp than "sub-range" -> `SubRange`

Reply:
What about `isInRange()`? Or `subrangeOf()`?

Or something like `range.intersect(otherRange) == range`?

Reply:
subrangeOf() sounds like it is returning a subrange

the intersect() version is just the inlined implementation without special case handling, because you need to use intersect2() and check for std::nullopt. Using a separate function **is** useful to reveal the intention!

Reply:
...and to avoid repetition.

Reply:
I have also used empty() instead of isEmpty() in the past, which unfortunately introduced many inconsistencies. Now I have to admit that being explicit is the better choice for the readability of the code.